lib/ansible/module_utils/powershell/Ansible.ModuleUtils.Legacy.psm1

```diff
@@ -310,6 +310,36 @@ Function Parse-Args($arguments, $supports_check_mode = $false)
             msg = "remote module does not support check mode"
         }
     }
+
+    # new exec changes has $complex_args as a [Dictionary`2[String], [Object]]
```

Comment:
I think if we switched most of the wrapper arg defs to IDictionary, this might not be necessary...

Reply:
The commit I made yesterday has moved back to Hashtable, the performance hit and code complexity was just too much. PSCore v6 as a `-AsHashtable` parameter for `ConvertTo-Json` and older hosts just need a PSCustomObject to Hashtable method which I've added.

Reply:
Anyway this was a comment around keeping backwards compatibility with modules, there are slight differences between a Dictionary/Hashtable enumerator that caused issue with win_dsc and win_scheduled_task.